### PR TITLE
Reset the database intially if it's empty without specifying the option

### PR DESCRIPTION
**Pull request by jld3103**

### CODE REVIEW

File: internal/postgres.go [hunk 1]
```diff
@@ -80,27 +80,48 @@ func CreateUsers(conn *pgx.Conn, users []string) error {
...
 
-	return b != 0, nil
+	return b, nil
 }
 
 func CheckUserExists(conn *pgx.Conn, user string) (bool, error) {
-	a := conn.QueryRow(context.Background(), fmt.Sprintf("SELECT COUNT(*) FROM pg_roles WHERE rolname='%s';", user))
+	a := conn.QueryRow(
+		context.Background(),
+		fmt.Sprintf("SELECT EXISTS(SELECT * FROM pg_roles WHERE rolname='%s');", user),
```
[the remark this answers is not in the document]
Comment (dmeijboom) [reply internal/postgres.go:100]: Attack vector is probably low but SQL injection is possible here (`rolname`)

File: internal/postgres.go [hunk 2]
```diff
@@ -80,27 +80,48 @@ func CreateUsers(conn *pgx.Conn, users []string) error {
...
+		return false, fmt.Errorf("failed to decode row: %w", err)
+	}
+
+	return b, nil
+}
+
+func CheckTableExists(conn *pgx.Conn, schema, name string) (bool, error) {
+	a := conn.QueryRow(
+		context.Background(),
+		fmt.Sprintf("SELECT EXISTS(SELECT FROM pg_tables WHERE schemaname = '%s' AND tablename = '%s');", schema, name),
```
Comment (dmeijboom) [inline internal/postgres.go:115]: Attack vector is probably low but SQL injection is possible here (`schemaname`, `tablename`)

**Review by dmeijboom — CHANGES_REQUESTED** [review]
I see that we're using `context.Background()` inside functions that pass it through to the database library. That's not great. Instead we should let these function accept `context.Context` as a parameter so it can be propagated properly.

File: cmd/apply.go [hunk 3]
```diff
@@ -93,6 +93,14 @@ func NewApplyCommand() *cobra.Command {
 				}
 			}
 
+			schemaMigrationsTableExists, err := internal.CheckTableExists(conn, "public", "schema_migrations")
+			if err != nil {
+				log.Fatalf("Failed to check if public.schema_migrations exists: %v", err)
+			}
+			if !schemaMigrationsTableExists {
+				resetDatabase = true
```
[the remark this answers is not in the document]
Comment (jld3103) [reply cmd/apply.go:101]: when trek is rolled out initially, you need to set the reset flag in order for trek to apply any migrations. this is quite annoying on local envs, so this just checks if the trek has ever applied any migrations via go-migrate (which creates that table) and if not it enables the resetting the database.

[the remark this answers is not in the document]
Comment (dmeijboom) [reply cmd/apply.go:101]: Check, but what does resetting mean in this case? Deleting all resources within a database?

[the remark this answers is not in the document]
Comment (jld3103) [reply cmd/apply.go:101]: usually yes and that is used for developing, but in this case we just use the reset to setup the database while it is already empty

File: internal/postgres.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jld3103) [reply internal/postgres.go:100]: yes sql injection is possible, but the passed values come either from trek directly or the config file that is written by the developer. i don't think we need to care about it here, but let me know if you think we should.

File: internal/postgres.go (hunk 2 above)
Comment (jld3103) [reply internal/postgres.go:115]: see above

**Review by dmeijboom — APPROVED** [review]

**Pull request merged**